compiler/AST/ForallStmt.cpp

```diff
@@ -290,6 +290,14 @@ ForallStmt* isForallLoopBody(Expr* expr) {
   return NULL;
 }
 
+// Return a const ForallStmt* if 'expr' is its loopBody.
+const ForallStmt* isConstForallLoopBody(const Expr* expr) {
+  if (const ForallStmt* pfs = toConstForallStmt(expr->parentExpr))
+    if (expr == pfs->loopBody())
+      return pfs;
```

Comment:
code duplication here with isForallLoopBody is unfortunate but at least it's short